XamarinCommunityToolkit/Behaviors/TextValidationBehavior.shared.cs

```diff
@@ -112,5 +114,21 @@ void OnRegexPropertyChanged()
             => regex = RegexPattern != null
                 ? new Regex(RegexPattern, RegexOptions)
                 : null;
+
+        string ReduceWhiteSpaces(string value)
```

Comment:
What does this do? Remove all whitespaces? Why not simply say `string.Replace(' ', '');`?

Reply:
OK, so yeah, `IsWhitespace` does more than just spaces. See [here](https://docs.microsoft.com/en-us/dotnet/api/system.char.iswhitespace?view=netcore-3.1#remarks). But looks like we can simply call [`Trim`](https://docs.microsoft.com/en-us/dotnet/api/system.string.trim?view=netcore-3.1) (see under Note to Callers)

Reply:
Forget I said anything. Trim is just begin and end of course

Reply:
> OK, so yeah, `IsWhitespace` does more than just spaces. See [here](https://docs.microsoft.com/en-us/dotnet/api/system.char.iswhitespace?view=netcore-3.1#remarks). But looks like we can simply call [`Trim`](https://docs.microsoft.com/en-us/dotnet/api/system.string.trim?view=netcore-3.1) (see under Note to Callers)

this method reduces spaces quantity to 1
Input: ```"____Hi there!_________Who are you?______"```
Output ```"_Hi there!_Who are you?_"```

``string.Split``` + ```string.Join``` won't work for us cuz this will remove spaces from start and end of the string.

Reply:
Will be good and very important to have some tests for this though. I also see some cases for removing spaces altogether or maybe replace it with another character.

Reply:
> Will be good and very important to have some tests for this though. I also see some cases for removing spaces altogether or maybe replace it with another character.

Hmm, could you please provide an example? I thought about adding such flag but didn't realize where it can be used.
Now if you want to remove all spaces from start and end of the string you can use the "Trim" option (As I use in email test-case).

```"____myemail@gmail.com____"``` will be updated to ```"myemail@gmail.com"```

But when do we need to remove all spaces?

Reply:
I mean, people have weird systems, right? And even if it's only for design purposes. Maybe you're showing barcode data or a phone number and you (or your product owner... ;)) wants to show it as one number but it is stored as `12 34 56 789`.

Or maybe you are showing a MAC address and you want to replace all spaces with colons or the other way around. But maybe we should wait for someone to actually have a need for these cases.

Reply:
> I mean, people have weird systems, right? And even if it's only for design purposes. Maybe you're showing barcode data or a phone number and you (or your product owner... ;)) wants to show it as one number but it is stored as `12 34 56 789`.
> 
> Or maybe you are showing a MAC address and you want to replace all spaces with colons or the other way around. But maybe we should wait for someone to actually have a need for these cases.

Got it, thanks.
Looks like we should expose one more flag "RemoveAllNonDigits"? :)